ddl/index.go

```diff
@@ -940,7 +942,60 @@ func (w *worker) addTableIndex(t table.Table, indexInfo *model.IndexInfo, reorgI
 	}
 	defer closeAddIndexWorkers(idxWorkers)
 
-	return w.buildKVRangesIndex(t, idxWorkers, job, reorgInfo)
+	finish := false
+	for !finish {
+		err := w.buildKVRangesIndex(t, idxWorkers, job, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		finish, err = w.updateReorgInfo(t, reorgInfo)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// updateReorgInfo will find the next partition according to current reorgInfo.
+func (w *worker) updateReorgInfo(t table.Table, reorg *reorgInfo) (bool, error) {
+	pi := t.Meta().GetPartitionInfo()
+	if pi == nil {
+		return true, nil
+	}
+
+	pid, err := findNextPartitionID(reorg.PartitionID, pi.Definitions)
+	if err != nil {
+		// Fatal error, should not run here.
+		log.Error("[ddl-reorg] update reorg fail, %v error stack: %s", t, errors.ErrorStack(err))
```

Comment:
I think we needn't the error stack here.  Then we can make this error log simpler.